Source/Orts.Simulation/Simulation/Signalling/SignalObject.cs

```diff
@@ -4223,18 +4224,38 @@ public bool[] RequestHoldSignalDispatcher(bool requestResetSignal)
                 {
                     signalRef.BreakDownRouteList(enabledTrain.Train.ValidRoute[enabledTrain.TrainRouteDirectionIndex], signalRouteIndex, enabledTrain);
                     ResetSignal(true);
-                    holdState = HoldState.ManualLock;
-                    returnValue[0] = true;
-                    returnValue[1] = true;
-                }
-                else //hopefully this does not happen
-                {
-                    holdState = HoldState.ManualLock;
-                    returnValue[0] = true;
                 }
+
+                RequestMostRestrictiveAspect();
             }
+        }
 
-            return returnValue;
+        public void RequestMostRestrictiveAspect()
+        {
+            holdState = HoldState.ManualLock;
+            foreach (var sigHead in SignalHeads)
+            {
+                sigHead.RequestMostRestrictiveAspect();
+            }
+        }
+
+        public void RequestApproachAspect()
+        {
+            holdState = HoldState.ManualApproach;
+            foreach (var sigHead in SignalHeads)
+            {
+                sigHead.RequestApproachAspect();
+            }
+        }
+
+        public void RequestLeastRestrictiveAspect()
+        {
+            holdState = HoldState.ManualPass;
+            foreach (var sigHead in SignalHeads)
+            {
+                sigHead.RequestLeastRestrictiveAspect();
+                sigHead.draw_state = sigHead.def_draw_state(sigHead.state);
```

Comment:
Why are you forcing the draw state here? This will override the draw aspect set by the C# script after the RequestLeastRestrictiveAspect event is received.

Reply:
I probably messed up something during rebase.
I have sent a fix.